Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -3061,6 +3061,29 @@ extension BrowserViewController: PreferencesObserver {
             setupTabs()
             updateTabsBarVisibility()
             updateApplicationShortcuts()
+            
+            // Destroy all tab data
+            tabManager.allTabs.forEach({
+                tabManager.removeAllBrowsingDataForTab($0, completionHandler: { [weak self] in
+                    self?.tabManager.removeAll()
```

Comment:
I don't get it, why we iterate over each tab, and for each tab we call `removeAll()` to remove all tabs here?

Reply:
Fixed.